src/flint/types/nmod_mpoly.pyx

```diff
@@ -0,0 +1,1276 @@
+                return res
+            else:
+                raise DomainError("nmod_mpoly division is not exact")
+        elif typecheck(other, nmod):
+            if other.modulus() != self.ctx.modulus():
+                raise ValueError(f"cannot divide with different modulus {other.modulus()} vs {self.ctx.modulus()}")
+            elif not other:
+                raise ZeroDivisionError("nmod_mpoly division by zero")
+            res = create_nmod_mpoly(self.ctx)
+            div = create_nmod_mpoly(self.ctx)
+            nmod_mpoly_set_ui(div.val, int(other), self.ctx.val)
+            if nmod_mpoly_divides(res.val, self.val, div.val, self.ctx.val):
+                return res
+            else:
+                raise DomainError("nmod_mpoly division is not exact")
+        else:
+            return NotImplemented
+
+    def __rtruediv__(self, other):
+        cdef nmod_mpoly res
+        if not self.ctx.is_prime():
+            raise DomainError("division with non-prime modulus is not supported")
+        elif not self:
+            raise ZeroDivisionError("nmod_mpoly division by zero")
+        elif isinstance(other, int):
+            res = create_nmod_mpoly(self.ctx)
+            div = create_nmod_mpoly(self.ctx)
+            nmod_mpoly_set_ui(div.val, other, self.ctx.val)
+            if nmod_mpoly_divides(res.val, div.val, self.val, self.ctx.val):
+                return res
+            else:
+                raise DomainError("nmod_mpoly division is not exact")
+        elif typecheck(other, nmod):
+            if other.modulus() != self.ctx.modulus():
+                raise ValueError(f"cannot divide with different modulus {other.modulus()} vs {self.ctx.modulus()}")
+            res = create_nmod_mpoly(self.ctx)
+            div = create_nmod_mpoly(self.ctx)
+            nmod_mpoly_set_ui(div.val, int(other), self.ctx.val)
+            if nmod_mpoly_divides(res.val, div.val, self.val, self.ctx.val):
+                return res
+            else:
+                raise DomainError("nmod_mpoly division is not exact")
+        else:
+            return NotImplemented
+
+    def __mod__(self, other):
+        return divmod(self, other)[1]
+
+    def __rmod__(self, other):
+        return divmod(other, self)[1]
+
+    def __call__(self, *args) -> ulong:
+        cdef:
+            slong nvars = self.ctx.nvars(), nargs = len(args)
+
+        if nargs < nvars:
+            raise ValueError("not enough arguments provided")
+        elif nargs > nvars:
+            raise ValueError("more arguments provided than variables")
+
+        args = [int(x) for x in args]
+
+        cdef:
+            # Using sizeof(ulong) here breaks on 64 windows machines because of the `ctypedef unsigned long ulong` in
+            # flintlib/flint.pxd. Cython will inline this definition and then allocate the wrong amount of memory.
+            ulong *vals = <ulong *>libc.stdlib.malloc(nargs * (FLINT_BITS // 4))
```

Comment:
Should this not be `FLINT_BITS // 8`?

Reply:
It certainly should, whoops! But I'll favour the suggested fix over this